docs/enhancements/on-prem-data-collection.md

```diff
@@ -0,0 +1,220 @@
+2. Allows flexibility for our team
+    - We can adjust the configurations on the fly such as where data is sent to
+    - E.g. if the original data collection endpoint is unavailable or if we choose to use a different service later on, we only need to change the configuration endpoint 
+
+#### Drawbacks
+1. Another deployment artifact to update and maintain
+2. Doesn't fully future-proof against the remote configuration's URL changes
+    - Although we can always have the end user change the URL by redeploying the service with the new URL set as an environment variable 
+
+### User Stories
+
+#### Story 1
+
+As an infrastructure operator, I can provision and manage my fleet of clusters knowing that data points are automatically collected and sent to the Assisted Installer team without having to do any additional configuration. I know my data will be safe and secure and the team will only collect data they need to improve the product. 
+
+#### Story 2
+
+As a developer on the assisted installer team, I can analyze the customer data to determine if a feature is worth implementing/keeping/improving. I know that the customer data is accurate and up-to-date. All of the data is parse-able and can be easily tailored to the graphs/visualizations that help my analysis.
+
+#### Story 3
+
+As a product owner, I can determine if the product is moving in the right direction based on the actual customer data. I can help determine which features and bug fixes to prioritize based on the data.
+
+### Implementation Details/Notes/Constraints
+
+Assisted Service changes:
+1. New environment variables to be added:
+    - `DISABLE_DATA_COLLECTION`: If set to `true`, it will disable this feature and thereby allow a customer to opt-out, otherwise the feature is on by default.
+    - `DATA_COLLECTION_URL`: The URL the data will be sent to. (Data format env var?)
+    - `REMOTE_CONFIG_URL`: The URL for the manifest with configuration data for on-prem data collection variables such as `DATA_COLLECTION_URL`
+2. Add data collection and sending logic
+    - Collecting and sending: querying the DB for events of that cluster, sanitizing the data, and sending it as a tar.gz to the `DATA_COLLECTION_URL` over HTTPS.
+    - When it's called:
+        1. A cluster reaches one of the following end stages: `error`, `cancelled`, `installed`
+            - Each of these cluster stages represent an end state and that the cluster attempted installation.
+    - Sending data is non-blocking and sent as best effort
+3. Add logic for authenticating to the `DATA_COLLECTION_URL` 
+
+### Risks and Mitigations
+
+#### Risk 1: Overusing Customer's Network & CPU
+
+On-premises users might have large-scale clusters so network usage and processor usage could be impacted depending on how large the data set is.
+
+##### Mitigation
+
+To avoid user workflow interruption, data collection, processing, and sending can happen as a concurrent thread. Data sending should be non-blocking and best effort.
```

Comment:
Still the concurrent thread consumes resources out of the container's process. Perhaps we should do a POC on what scales are involved?